internal/app/singularity/cache_list_linux.go

```diff
@@ -0,0 +1,222 @@
+	if err != nil {
+		sylog.Warningf("Unable to opening cache folder: %v", err)
+		return err
+	}
+	for _, f := range libraryCacheFiles {
+		cont, err := ioutil.ReadDir(filepath.Join(cache.Library(), f.Name()))
+		if err != nil {
+			sylog.Warningf("Unable to looking in cache: %v", err)
+			return err
+		}
+		for _, c := range cont {
+			fileInfo, err := os.Stat(filepath.Join(cache.Library(), f.Name(), c.Name()))
+			if err != nil {
+				sylog.Warningf("Unable to get stat: %v", err)
+				return err
+			}
+			printFileSize, err := findSize(fileInfo.Size())
+			if err != nil {
+				// no need to describe the error, since it is already
+				sylog.Warningf("%v", err)
+			}
+			fmt.Printf("%-22s %-22s %-16s %s\n", c.Name(), fileInfo.ModTime().Format("2006-01-02 15:04:05"), printFileSize, "library")
+			printFileSize = ""
+		}
+	}
+	return nil
+}
+
+func listOciCache() error {
+	// loop thrught oci-tmp cache
+	ociTmp, err := ioutil.ReadDir(cache.OciTemp())
+	if err != nil {
+		sylog.Warningf("Unable to opening oci-tmp folder: %v", err)
+		return err
+	}
+	for _, f := range ociTmp {
+		blob, err := ioutil.ReadDir(filepath.Join(cache.OciTemp(), f.Name()))
+		if err != nil {
+			sylog.Warningf("Unable to looking in cache: %v", err)
+			return err
+		}
+		for _, b := range blob {
+			fileInfo, err := os.Stat(filepath.Join(cache.OciTemp(), f.Name(), b.Name()))
+			if err != nil {
+				sylog.Warningf("Unable to get stat: %v", err)
+				return err
+			}
+			printFileSize, err := findSize(fileInfo.Size())
+			if err != nil {
+				// no need to describe the error, since it is already
+				sylog.Warningf("%v", err)
+			}
+			fmt.Printf("%-22s %-22s %-16s %s\n", b.Name(), fileInfo.ModTime().Format("2006-01-02 15:04:05"), printFileSize, "oci")
+			printFileSize = ""
+		}
+	}
+	return nil
+}
+
+func listBlobCache(printList bool) error {
+	// loop thrught ociBlob cache
+	count := 0
+	var totalSize int64
+
+	_, err = os.Stat(filepath.Join(cache.OciBlob(), "/blobs"))
+	if os.IsNotExist(err) {
+		return nil
+	}
+	blobs, err := ioutil.ReadDir(filepath.Join(cache.OciBlob(), "/blobs/"))
+	if err != nil {
+		sylog.Warningf("Unable to opening oci folder: %v", err)
+		return err
+	}
+	for _, f := range blobs {
+		blob, err := ioutil.ReadDir(filepath.Join(cache.OciBlob(), "/blobs/", f.Name()))
+		if err != nil {
+			sylog.Warningf("Unable to looking in cache: %v", err)
+			return err
+		}
+		for _, b := range blob {
+			fileInfo, err := os.Stat(filepath.Join(cache.OciBlob(), "/blobs/", f.Name(), b.Name()))
+			if err != nil {
+				sylog.Warningf("Unable to get stat: %v", err)
+				return err
+			}
+			if printList == true {
+				printFileSize, err := findSize(fileInfo.Size())
+				if err != nil {
+					// no need to describe the error, since it is already
+					sylog.Warningf("%v", err)
+				}
+				fmt.Printf("%-22.20s %-22s %-16s %s\n", b.Name(), fileInfo.ModTime().Format("2006-01-02 15:04:05"), printFileSize, "blob")
+				printFileSize = ""
```

Comment:
I'm not quite sure why we're ding this assignment here. In Go, variable declaration/assignments are namespaced in a way. So, by doing `printFileSize, err := findSize(...)` inside of this `if printList` block, we don't have to re-assign it to an empty value. The `:=` assignment/declaration operator will assure that every time we enter into that if block, it will be a blank copy

Reply:
Opps! I meant to delete this line :man_facepalming: